### PR TITLE
Filter double appearing addressbooks

### PR DESCRIPTION
**Pull request by Julian1998**
## Description

Filter for the double-appearing address book, when the user is sharing an addressbook with groups he is also a member of.

## How Has This Been Tested?

- Create an addressbook in contacts app
- Share it with a group the user also belongs to
- The shared addressbook is now filtered

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised:

## Open tasks:

- [ ] Backport

### CODE REVIEW

**Comment by Julian1998** [comment]
I still have to modify CardDavBackend unit tests.

File: apps/dav/lib/CardDAV/CardDavBackend.php [hunk 1]
```diff
@@ -162,7 +162,7 @@ public function getAddressBooksForUser($principalUri) {
 		$principals = $this->principalBackend->getGroupMembership($principalUri, true);
 		$principals[]= $principalUri;
 
-		$addressBooksIds = [];
+		$addressBooksIds[] = -1;
```
Comment (Julian1998) [inline apps/dav/lib/CardDAV/CardDavBackend.php]: DB query fails if addressBooksIds is empty, so if the user doesn't have any addressbooks. This is the fix but rly dirty. Any suggestions how to do better @DeepDiver1975

Comment (DeepDiver1975) [reply apps/dav/lib/CardDAV/CardDavBackend.php]: de-duplication should happen a few lines below .... L187

Comment (Julian1998) [reply apps/dav/lib/CardDAV/CardDavBackend.php]: @DeepDiver1975 I see the problem in the db query! Why ask for more information than needed and afterwards filter that infos again? It's unneccessary in my opinion and we could save performance by doing it there

**Comment by VicDeo** [comment]
@Julian1998 looks good in general, please rebase

**Comment by DeepDiver1975** [comment]
> @Julian1998 looks good in general, please rebase

@VicDeo feel free to grab the PR - @Julian1998 is only working 1 day per week - usually fridays ....

**Review by VicDeo — APPROVED** [review]
Rebased, squashed.
Tested and works.

**Comment by VicDeo** [comment]
Stable10: https://github.com/owncloud/core/pull/35603

**Pull request merged**